src/core/server/http/http_tools.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { readFileSync } from 'fs';
+import { Server, ServerOptions } from 'hapi-latest';
+import { ServerOptions as TLSOptions } from 'https';
+import { HttpConfig } from './http_config';
+
+/**
+ * Converts Kibana `HttpConfig` into `ServerOptions` that are accepted by the Hapi server.
+ */
+export function getServerOptions(
+  config: HttpConfig,
+  { configureTLS = true } = {}
+) {
+  // Note that all connection options configured here should be exactly the same
+  // as in the legacy platform server (see `server/http/index`). Any change
```

Comment:
can we be super clear please and say src/server/http/index ?

Reply:
and also, can we be super clear in src/server/http/index.js on line 39, and say src/core/server/http/http_tools?

Reply:
> can we be super clear please and say src/server/http/index ?
> and also, can we be super clear in src/server/http/index.js on line 39, and say src/core/server/http/http_tools?

Sure!